### ISSUE TITLE
om upload-product can show a failure if Ops Manager takes too long to respond 

### ISSUE THREAD

**Issue by pivotal-todd-ritchie** [pivotal-cf/om#366]
`om` can show a failure when uploading large tiles that take a long time to decompress. As an example, I'm having trouble uploading SRT on a brand new terraformed azure opsman.

When I upload the tile, ops manager takes little over 6 minutes to decompress and validate the tile. By default, the idle connection timeout on Azure is 4 minutes. 

The result looks like this:

```
+ om -k upload-product --product ./srt-2.5.2-build.7.pivotal
processing product
beginning product upload to Ops Manager
 10.12 GiB / 10.12 GiB  100.00% 8m9s1s2s
21m50s elapsed, waiting for response from Ops Manager...
retrying product upload after error: could not make api request to available_products endpoint: Post https://pcf.example.com/api/v0/available_products: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
processing product
beginning product upload to Ops Manager
...
```

If I browse to the ops manager about 10 minutes into the "waiting for response from Ops Manager" I can see the tile is available to stage. `om` sees the timeout, retries twice, and then exits with a failure after exhausting retries.

My workaround is to increase the idle connection timeout on the Azure VM. Having `om` check if the tile is there already before each retry attempt could also help.

**Comment by kcboyle** [comment]
The PR has been prioritized for review.
Steps our team will go through before/after pulling it in:
- include a concrete procedure for replicating the error
- affirm that we do not error if the product is already there

To our team's understanding, this is most easily replicate-able on Azure, and is due to a server timeout on the Azure side, _not_ a timeout from `om`.

**Comment by anEXPer** [comment]
We're pretty sure this isn't governed by the Azure idle timeout, but by our default `request-timeout`.

The upshot is: extend your request-timeout in your env.yml file and you won't see this problem. We've heard that "over 9000" is sufficient, but probably excessive.

We've also updated the related PR with more information about changes we'd like to see in the messaging, but increasing the timeout is probably the best play for most people, as those changes may not land for a while.

**Comment by anEXPer** [comment]
Here's our followup story: https://www.pivotaltracker.com/story/show/166833456

**Comment by kcboyle** [comment]
@pivotal-todd-ritchie this should have been released with better error messaging in v3.0.0. 

Closing this issue.
